File: cadwiki-nuget/cadwiki.NUnitTestRunner/Creators/HtmlCreator.vb
Imports System.IO
Imports System.Reflection
Imports HandlebarsDotNet

Public Class HtmlCreator
    Public Sub ParameterizeReportTemplate()
        Dim assembly As Assembly = System.Reflection.Assembly.GetExecutingAssembly()
        Dim templateString As String = NetUtils.AssemblyUtils.ReadEmbeddedResourceToString(assembly, "HtmlTemplateTestSuiteReport.cshtml")
        If (templateString IsNot Nothing) Then
            Dim template As HandlebarsDotNet.HandlebarsTemplate(Of Object, Object) = Handlebars.Compile(templateString)
            Dim reportModel As New HtmlModelTestSuiteReport
            reportModel.BannerImagePath = "test"
            Dim result As Object = template(reportModel)
        End If
    End Sub
End Class

File: cadwiki-nuget/cadwiki.NUnitTestRunner/Creators/HtmlCreator.cs
using System.IO;
using System.Reflection;
using HandlebarsDotNet;

namespace cadwiki.NUnitTestRunner
{

    public class HtmlCreator
    {

        private HandlebarsTemplate<object, object> _testSuiteReportTemplateToObject;
        private string _parametizedReport;
        public HtmlCreator()
        {
            var @assembly = Assembly.GetExecutingAssembly();
            CompileAllTemplates(assembly);
        }

        private void CompileAllTemplates(Assembly @assembly)
        {
            string testSuiteReportTemplate = NetUtils.AssemblyUtils.ReadEmbeddedResourceToString(assembly, "TestSuiteReport.html");
            _testSuiteReportTemplateToObject = Handlebars.Compile(testSuiteReportTemplate);
        }

        public string ParameterizeReportTemplate(HTML_Models.TestSuiteReport model)
        {
            if (_testSuiteReportTemplateToObject is not null)
            {
                _parametizedReport = _testSuiteReportTemplateToObject(model);
                return _parametizedReport;
            }
            return null;
        }

        public void SaveReportToFile(string filePath)
        {
            filePath = NetUtils.Paths.GetUniqueFilePath(filePath);
            File.WriteAllText(filePath, _parametizedReport);
        }

    }
}